examples/viewer.py

```diff
@@ -0,0 +1,1489 @@
+                self.replay_renderer.set_sensor_transform(
+                    i, sensor_uuid, transform
+                )
+        # Render
+        self.replay_renderer.render(mn.gl.default_framebuffer)
+
+    def move_and_look(self, repetitions: int) -> None:
+        """
+        This method is called continuously with `self.draw_event` to monitor
+        any changes in the movement keys map `Dict[KeyEvent.key, Bool]`.
+        When a key in the map is set to `True` the corresponding action is taken.
+        """
+        # avoids unnecessary updates to grabber's object position
+        if repetitions == 0:
+            return
+
+        key = Application.KeyEvent.Key
+        agent = self.sim.agents[self.agent_id]
+        press: Dict[key.key, bool] = self.pressed
+        act: Dict[key.key, str] = self.key_to_action
+
+        action_queue: List[str] = [act[k] for k, v in press.items() if v]
+
+        for _ in range(int(repetitions)):
+            [agent.act(x) for x in action_queue]
+
+        # update the grabber transform when our agent is moved
+        if self.mouse_grabber is not None:
+            # update location of grabbed object
+            self.update_grab_position(self.previous_mouse_point)
+
+    def invert_gravity(self) -> None:
+        """
+        Sets the gravity vector to the negative of it's previous value. This is
+        a good method for testing simulation functionality.
+        """
+        gravity: mn.Vector3 = self.sim.get_gravity() * -1
+        self.sim.set_gravity(gravity)
+
+    def key_press_event(self, event: Application.KeyEvent) -> None:
+        """
+        Handles `Application.KeyEvent` on a key press by performing the corresponding functions.
+        If the key pressed is part of the movement keys map `Dict[KeyEvent.key, Bool]`, then the
+        key will be set to False for the next `self.move_and_look()` to update the current actions.
+        """
+        key = event.key
+        pressed = Application.KeyEvent.Key
+        mod = Application.InputEvent.Modifier
+
+        shift_pressed = bool(event.modifiers & mod.SHIFT)
+        alt_pressed = bool(event.modifiers & mod.ALT)
+        # warning: ctrl doesn't always pass through with other key-presses
+
+        if key == pressed.ESC:
+            event.accepted = True
+            self.exit_event(Application.ExitEvent)
+            return
+
+        elif key == pressed.H:
+            self.print_help_text()
+        elif key == pressed.J:
+            logger.info(
+                f"Toggle Region Draw from {self.semantic_region_debug_draw } to {not self.semantic_region_debug_draw}"
+            )
+            # Toggle visualize semantic bboxes. Currently only regions supported
+            self.semantic_region_debug_draw = (
+                not self.semantic_region_debug_draw
+            )
+
+        elif key == pressed.TAB:
+            # NOTE: (+ALT) - reconfigure without cycling scenes
+            # if not alt_pressed:
```

Comment:
this commenting is intentional? :)

Reply:
Yeah, I cannibalized an existing app and changed functionality. ALT+TAB won't work anyway since it is reserved by system OS or window switch. :laughing: